testkit/core/src/main/scala/com/lightbend/lagom/internal/testkit/TestConfig.scala

```diff
@@ -5,21 +5,29 @@ package com.lightbend.lagom.internal.testkit
 
 import com.typesafe.config.{ Config, ConfigFactory }
 
+import scala.util.Try
+
 private[lagom] object TestConfig {
-  lazy val JdbcConfig: Config = ConfigFactory.parseString(
-    """
-      |db.default.driver=org.h2.Driver
-      |db.default.url="jdbc:h2:mem:service-test"
-      |
-      |jdbc-defaults.slick.profile = "slick.jdbc.H2Profile$"
-      |
-      |jdbc-journal.slick.profile = ${?jdbc-defaults.slick.profile}
-      |jdbc-read-journal.slick.profile = ${?jdbc-defaults.slick.profile}
-      |jdbc-snapshot-store.slick.profile = ${?jdbc-defaults.slick.profile}
-      |lagom.persistence.read-side.jdbc.slick.profile = ${?jdbc-defaults.slick.profile}
-      |
-      |akka.persistence.journal.plugin = jdbc-journal
-      |akka.persistence.snapshot-store.plugin = jdbc-snapshot-store
+  private lazy val userConfigs = ConfigFactory.load()
+
+  private def getConfig(key: String, fallbackValue: String): String =
+    s"""$key = "${Try(userConfigs.getString(key)).getOrElse(fallbackValue)}""""
+
+  private lazy val values =
+    s"""
+       |${getConfig("db.default.driver", "org.h2.Driver")}
+       |${getConfig("db.default.url", "jdbc:h2:mem:service-test")}
+       |
+       |${getConfig("jdbc-defaults.slick.profile", "slick.jdbc.H2Profile$")}
```

Comment:
We could improve the design of this code by reducing the possibilities:

If we found a value for `jdbc-defaults.slick.profile` key in `userConfigs`
then we use `userConfigs` values for `jdbc-defaults.slick.profile`, `db.default.driver`, and `db.default.url`
else we use the "fallback" values (H2)

Could you be interested in such design ?

Reply:
The code in `getConfig(...)` seems similar to how loading `application.conf` and `reference.conf` works. The problem with `reference.conf` is that it's loaded in all cases and we want to programmatically control what `journal` and `snapshot` plugin is used for tests.

I wonder if we could achieve the same result moving the `jdbc`-specific settings to `testkit/core/src/main/resources/reference.conf`. That is, create a new `testkit/core/src/main/resources/reference.conf` with:

```
db.default.driver=org.h2.Driver
db.default.url="jdbc:h2:mem:service-test"

jdbc-defaults.slick.profile = "slick.jdbc.H2Profile$"

jdbc-journal.slick.profile = ${?jdbc-defaults.slick.profile}
jdbc-read-journal.slick.profile = ${?jdbc-defaults.slick.profile}
jdbc-snapshot-store.slick.profile = ${?jdbc-defaults.slick.profile}
lagom.persistence.read-side.jdbc.slick.profile = ${?jdbc-defaults.slick.profile}
```

but keep `TestConfig` with:

```
  lazy val JdbcConfig: Config = ConfigFactory.parseString(
    """
      |akka.persistence.journal.plugin = jdbc-journal
      |akka.persistence.snapshot-store.plugin = jdbc-snapshot-store
    """.stripMargin
  ).resolve()
```

Do you think it'd work?

Reply:
It was my first idea to move these settings in a `application.conf` or `reference.conf` because it's how `typesafe-config` works (I'm not a expert at all of how this lib works) and I don't want to reinvent the wheel but I wasn't sure it would work and the actual PR changes less thing.

I'll try in another branch 😉